test/tstMinimumSpanningTree.cpp

```diff
@@ -0,0 +1,168 @@
+
+} // namespace Test
+
+BOOST_AUTO_TEST_CASE_TEMPLATE(minimum_spanning_tree, DeviceType,
+                              ARBORX_DEVICE_TYPES)
+{
+  using ExecutionSpace = typename DeviceType::execution_space;
+  ExecutionSpace exec_space;
+
+  { // equidistant points
+    // 0     1     2     3     4
+    //[0]   [1]   [2]   [3]   [4]
+    std::vector<ArborX::Point> points{
+        {0, 0, 0}, {1, 0, 0}, {2, 0, 0}, {3, 0, 0}, {4, 0, 0},
+    };
+
+    ARBORX_TEST_MINIMUM_SPANNING_TREE(
+        exec_space, points, 1,
+        (std::vector<ArborX::Details::WeightedEdge>{
+            {0, 1, 1}, {1, 2, 1}, {2, 3, 1}, {3, 4, 1}}));
+
+    ARBORX_TEST_MINIMUM_SPANNING_TREE(
+        exec_space, points, 2,
+        (std::vector<ArborX::Details::WeightedEdge>{
+            {0, 1, 1}, {1, 2, 1}, {2, 3, 1}, {3, 4, 1}}));
+
+    ARBORX_TEST_MINIMUM_SPANNING_TREE(
+        exec_space, points, 3,
+        (std::vector<ArborX::Details::WeightedEdge>{
+            {0, 1, 2}, {1, 2, 1}, {2, 3, 1}, {2, 4, 2}}));
+
+    ARBORX_TEST_MINIMUM_SPANNING_TREE(
+        exec_space, points, 4,
+        (std::vector<ArborX::Details::WeightedEdge>{
+            {0, 1, 3}, {1, 2, 2}, {1, 3, 2}, {1, 4, 3}}));
+
+    ARBORX_TEST_MINIMUM_SPANNING_TREE(
+        exec_space, points, 5,
+        (std::vector<ArborX::Details::WeightedEdge>{
+            {0, 1, 4}, {1, 2, 3}, {1, 3, 3}, {0, 4, 4}}));
+  }
+  { // non-equidistant points
+    // 0   1   2   3   4   5   6   7   8   9   10
+    //[0] [1] [2] [3]         [4]             [5]
+    std::vector<ArborX::Point> non_equidistant_points{
+        {0, 0, 0}, {1, 0, 0}, {2, 0, 0}, {3, 0, 0}, {6, 0, 0}, {10, 0, 0},
+    };
+
+    ARBORX_TEST_MINIMUM_SPANNING_TREE(
+        exec_space, non_equidistant_points, 1,
+        (std::vector<ArborX::Details::WeightedEdge>{
+            {0, 1, 1}, {1, 2, 1}, {2, 3, 1}, {3, 4, 3}, {4, 5, 4}}));
+
+    ARBORX_TEST_MINIMUM_SPANNING_TREE(
+        exec_space, non_equidistant_points, 2,
+        (std::vector<ArborX::Details::WeightedEdge>{
+            {0, 1, 1}, {1, 2, 1}, {2, 3, 1}, {3, 4, 3}, {4, 5, 4}}));
+
+    ARBORX_TEST_MINIMUM_SPANNING_TREE(
+        exec_space, non_equidistant_points, 3,
+        (std::vector<ArborX::Details::WeightedEdge>{
+            {0, 1, 2}, {1, 2, 1}, {1, 3, 2}, {2, 4, 4}, {3, 5, 7}}));
+
+    ARBORX_TEST_MINIMUM_SPANNING_TREE(
+        exec_space, non_equidistant_points, 4,
+        (std::vector<ArborX::Details::WeightedEdge>{
+            {0, 1, 3}, {1, 2, 2}, {0, 3, 3}, {2, 4, 4}, {2, 5, 8}}));
+
+    ARBORX_TEST_MINIMUM_SPANNING_TREE(
+        exec_space, non_equidistant_points, 5,
+        (std::vector<ArborX::Details::WeightedEdge>{
+            {0, 1, 6}, {1, 2, 5}, {2, 3, 4}, {1, 4, 5}, {1, 5, 9}}));
+
+    ARBORX_TEST_MINIMUM_SPANNING_TREE(
+        exec_space, non_equidistant_points, 6,
+        (std::vector<ArborX::Details::WeightedEdge>{
+            {0, 1, 10}, {1, 2, 9}, {2, 3, 8}, {3, 4, 7}, {0, 5, 10}}));
+  }
```

Comment:
Not blocking comment.

I think we should add a larger 3D to get some confidence. I looked around, and found that MLPACK [tests](https://github.com/mlpack/mlpack/blob/83e70110595eaf3cf3758f270433801e673615b2/src/mlpack/tests/emst_test.cpp) Euclidean MST for a [dataset]() of 1000 3D points. We could run MLPACK, dump its result, and use both dataset and the result for validation.

Reply:
Totally fine, don't want to hold this PR up.